src/cascadia/TerminalSettingsModel/KeyMapping.cpp

```diff
@@ -141,4 +141,40 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 
         return keyModifiers;
     }
+
+    // Method Description:
+    // - Build a map of all the globalSummon actions.
+    // - quakeMode actions are included in this, but expanded to the equivalent
+    //   set of GlobalSummonArgs
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - a map of KeyChord -> ActionAndArgs containing all globally bindable actions.
+    Windows::Foundation::Collections::IMap<Control::KeyChord, Model::ActionAndArgs> KeyMapping::FetchGlobalHotkeys()
+    {
+        std::unordered_map<Control::KeyChord, Model::ActionAndArgs, KeyChordHash, KeyChordEquality> justGlobals;
+
+        for (const auto& [k, v] : _keyShortcuts)
+        {
+            if (v.Action() == ShortcutAction::GlobalSummon)
+            {
+                justGlobals[k] = v;
+            }
+            else if (v.Action() == ShortcutAction::QuakeMode)
+            {
+                // Manually replace the QuakeMode action with a globalSummon
+                // that has the appropriate action args.
+                auto args = winrt::make_self<GlobalSummonArgs>();
+
+                // We want to summon the window with the name "_quake" specifically.
+                args->Name(L"_quake");
+
+                Model::ActionAndArgs actionAndArgs{ ShortcutAction::GlobalSummon, *args };
+                justGlobals[k] = actionAndArgs;
+            }
+        }
+
+        return winrt::single_threaded_map<Control::KeyChord, Model::ActionAndArgs>(std::move(justGlobals));

```

Comment:
Idk how often you call `FetchGlobalHotkeys`, but if you call it more than once, maybe you should consider caching it?

Reply:
It only happens in two scenarios:
* on becoming the monarch (which only happens once per window)
* when the settings reload (and the cache would inevitably be dirty)

So I didn't think this was much of an issue, but I can leave a note